src/lib/containers/widgets/facet-nav/FacetNav.tsx

```diff
@@ -0,0 +1,259 @@
+    )
+    if (!result) {
+      return
+    }
+    if (isFirstTime) {
+      setFacetUiStateArray(
+        result.map((item, index) => ({
+          name: item.columnName,
+          isHidden: index < 4 ? false : true,
+          isExpanded: false,
+        })),
+      )
+      setIsFirstTime(false)
+    }
+    setFacets(result)
+  }, [data])
+
+  // when 'show more/less' is clicked
+  const showMore = (shouldShowMore: boolean) => {
+    if (shouldShowMore) {
+      setFacetUiStateArray(facetUiStateArray =>
+        facetUiStateArray.map(item => {
+          return { ...item, isHidden: false }
+        }),
+      )
+    } else {
+      setFacetUiStateArray(facetUiStateArray =>
+        facetUiStateArray.map((item, index) =>
+          index >= 4 ? { ...item, isHidden: true } : { ...item },
+        ),
+      )
+    }
+  }
+
+  // what needs to happen after the filters are adjusted from the plot
+  const applyChangesFromQueryFilter = (facets: FacetColumnRequest[]) => {
+    request.query.selectedFacets = facets
+    executeQueryRequest!(request)
+  }
+
+  // don't show expanded or hidden facets
+  const isFacetHiddenInGrid = (columnName: string) => {
+    const itemHidden = facetUiStateArray.find(
+      item =>
+        item.name === columnName &&
+        (item.isHidden === true || item.isExpanded === true),
+    )
+    const result = itemHidden !== undefined
+    return result
+  }
+
+  const getShowMoreState = (): ShowMoreState => {
+    if (facetUiStateArray.length <= 4) {
+      return 'NONE'
+    }
+    if (
+      facetUiStateArray.find(item => item.isHidden === true) ||
+      facets.length < facetUiStateArray.length
+    ) {
+      return 'MORE'
+    }
+    return 'LESS'
+  }
+
+  // hides expanded facet under 'show more'
+  const hideExpandedFacet = (facet: FacetColumnResult, index: number) => {
+    setFacetUiStateArray(facetUiStateArray =>
+      facetUiStateArray.map(item =>
+        item.name === facet.columnName
+          ? { ...item, isExpanded: false, isHidden: true }
+          : item,
+      ),
+    )
+    setExpandedFacets(expandedFacets =>
+      expandedFacets.filter(item => item.facet.columnName != facet.columnName),
+    )
+  }
+
+  //expands to collapses a facet
+  const toggleExpandFacet = (
+    facet: FacetColumnResult,
+    index: number,
+    doExpand: boolean,
+  ) => {
+    setUiPropertyForFacet(facet.columnName, 'isExpanded', doExpand)
+    if (doExpand) {
+      setExpandedFacets([...expandedFacets, { facet, index }])
+    } else {
+      setExpandedFacets(
+        expandedFacets.filter(
+          item => item.facet.columnName != facet.columnName,
+        ),
+      )
+    }
+  }
+
+  // hides facet graph
+  const hideFacetInGrid = (columnName: string) => {
+    setUiPropertyForFacet(columnName, 'isHidden', true)
+  }
+
+  const setUiPropertyForFacet = (
+    columnName: string,
+    propName: 'isHidden' | 'isExpanded',
+    value: boolean,
+  ) => {
+    const cloneArray = _.cloneDeep(facetUiStateArray)
```

Comment:
What do lines 163-167 do? It looks like they could be removed

Reply:
Nice catch! Totally I reworked this logic and forgot to remove old implementation :see_no_evil: